### ISSUE TITLE
[Syntaxnet,dragnn] bazel test ... not passed all on Ubuntu 16.04 

### ISSUE THREAD

**Issue by anhphamm** [tensorflow/models#1232]
## [Syntaxnet,dragnn] bazel test ... not passed all on Ubuntu 16.04

Hi all,
I'm new to tensorflow and syntaxnet. I followed several Issues' responses but didn't solve the problems.

- Environment:
```
Linux kernel: Linux usar 4.4.0-66-generic #87-Ubuntu SMP Fri Mar 3 15:29:05 UTC 2017 x86_64 x86_64 x86_64 GNU/Linux
GCC:   gcc (Ubuntu 5.4.0-6ubuntu1~16.04.4) 5.4.0 20160609
G++:    g++ (Ubuntu 5.4.0-6ubuntu1~16.04.4) 5.4.0 20160609
Cuda: Cuda compilation tools, release 8.0, V8.0.44
$LD_LIBRARY_PATH :/usr/local/cuda/lib64:/usr/local/cuda/extras/CUPTI/lib64: No such file or directory
$CUDA_HOME /usr/local/cuda
python: Python 2.7.13 :: Anaconda custom (64-bit)
tensorflow/models: commit 0bd8e24d82c2f330b91767397da52072d477ab17

Tensorflow configure: 
All default except yes with CUDA option
```

 `bazel test ...`

```
INFO: Elapsed time: 3.855s, Critical Path: 3.52s
//dragnn/python:visualization_test                              (cached) PASSED in 2.9s
//syntaxnet:arc_standard_transitions_test                       (cached) PASSED in 0.1s
//syntaxnet:beam_reader_ops_test                                (cached) PASSED in 15.0s
//syntaxnet:binary_segment_state_test                           (cached) PASSED in 0.1s
//syntaxnet:binary_segment_transitions_test                     (cached) PASSED in 0.1s
//syntaxnet:char_ngram_string_extractor_test                    (cached) PASSED in 0.0s
//syntaxnet:char_properties_test                                (cached) PASSED in 0.1s
//syntaxnet:char_shift_transitions_test                         (cached) PASSED in 0.1s
//syntaxnet:graph_builder_test                                  (cached) PASSED in 7.5s
//syntaxnet:head_transitions_test                               (cached) PASSED in 0.2s
//syntaxnet:label_transitions_test                              (cached) PASSED in 0.1s
//syntaxnet:lexicon_builder_test                                (cached) PASSED in 1.6s
//syntaxnet:morphology_label_set_test                           (cached) PASSED in 0.0s
//syntaxnet:once_transitions_test                               (cached) PASSED in 0.1s
//syntaxnet:parser_features_test                                (cached) PASSED in 0.1s
//syntaxnet:parser_trainer_test                                 (cached) PASSED in 23.0s
//syntaxnet:reader_ops_test                                     (cached) PASSED in 3.2s
//syntaxnet:segmenter_utils_test                                (cached) PASSED in 0.0s
//syntaxnet:sentence_features_test                              (cached) PASSED in 0.1s
//syntaxnet:shared_store_test                                   (cached) PASSED in 0.3s
//syntaxnet:tagger_transitions_test                             (cached) PASSED in 0.1s
//syntaxnet:text_formats_test                                   (cached) PASSED in 1.8s
//syntaxnet/util:check_test                                     (cached) PASSED in 1.7s
//syntaxnet/util:registry_test                                  (cached) PASSED in 1.8s
//syntaxnet:whole_sentence_features_test                        (cached) PASSED in 0.1s
//util/utf8:unicodetext_unittest                                (cached) PASSED in 0.0s
//dragnn/components/syntaxnet:syntaxnet_component_test                NO STATUS
//dragnn/components/syntaxnet:syntaxnet_link_feature_extractor_test   NO STATUS
//dragnn/components/syntaxnet:syntaxnet_transition_state_test         NO STATUS
//dragnn/core:compute_session_impl_test                               NO STATUS
//dragnn/core:compute_session_pool_test                               NO STATUS
//dragnn/core:dragnn_bulk_op_kernels_test                             NO STATUS
//dragnn/core:dragnn_op_kernels_test                                  NO STATUS
//dragnn/core:index_translator_test                                   NO STATUS
//dragnn/core:input_batch_cache_test                                  NO STATUS
//dragnn/core:resource_container_test                                 NO STATUS
//dragnn/io:sentence_input_batch_test                                 NO STATUS
//dragnn/python:bulk_component_test                                   NO STATUS
//dragnn/python:composite_optimizer_test                              NO STATUS
//dragnn/python:digraph_ops_test                                      NO STATUS
//dragnn/python:evaluation_test                                       NO STATUS
//dragnn/python:graph_builder_test                                    NO STATUS
//dragnn/python:lexicon_test                                          NO STATUS
//dragnn/python:network_units_test                                    NO STATUS
//dragnn/python:render_parse_tree_graphviz_test                       NO STATUS
//dragnn/python:render_spec_with_graphviz_test                        NO STATUS
//dragnn/python:sentence_io_test                                      NO STATUS
//dragnn/python:spec_builder_test                                     NO STATUS
//examples/dragnn:test_run_all_tutorials                              NO STATUS

Executed 0 out of 50 tests: 26 tests pass, 1 fails to build and 23 were skipped.
There were tests whose specified size is too big. Use the --test_verbose_timeout_warnings command line option to see which ones these are.
```

```
INFO: Found 162 targets and 50 test targets...
ERROR: /home/usar/ws/pretrained/models/syntaxnet/dragnn/core/BUILD:102:1: C++ compilation of rule '//dragnn/core:beam_test' failed: gcc failed: error executing command /usr/bin/gcc -U_FORTIFY_SOURCE -fstack-protector -Wall -B/usr/bin -B/usr/bin -Wunused-but-set-parameter -Wno-free-nonheap-object -fno-omit-frame-pointer -g0 -O2 '-D_FORTIFY_SOURCE=1' -DNDEBUG ... (remaining 125 argument(s) skipped): com.google.devtools.build.lib.shell.BadExitStatusException: Process exited with status 1.
In file included from external/org_tensorflow/tensorflow/core/platform/default/logging.h:24:0,
                 from external/org_tensorflow/tensorflow/core/platform/logging.h:25,
                 from external/org_tensorflow/tensorflow/core/lib/core/status.h:24,
                 from ./syntaxnet/base.h:24,
                 from ./dragnn/core/interfaces/transition_state.h:7,
                 from ./dragnn/core/interfaces/cloneable_transition_state.h:7,
                 from ./dragnn/core/beam.h:8,
                 from dragnn/core/beam_test.cc:1:
./dragnn/core/beam.h: In member function 'void syntaxnet::dragnn::Beam<T>::AdvanceFromPrediction(const float*, int, int)':
./dragnn/core/beam.h:115:37: error: there are no arguments to 'isnan' that depend on a template parameter, so a declaration of 'isnan' must be available [-fpermissive]
             CHECK(!isnan(score_delta));
                                     ^
external/org_tensorflow/tensorflow/core/platform/macros.h:64:30: note: in definition of macro 'TF_PREDICT_FALSE'
 #define TF_PREDICT_FALSE(x) (x)
                              ^
./dragnn/core/beam.h:115:13: note: in expansion of macro 'CHECK'
             CHECK(!isnan(score_delta));
             ^
./dragnn/core/beam.h:115:37: note: (if you use '-fpermissive', G++ will accept your code, but allowing the use of an undeclared name is deprecated)
             CHECK(!isnan(score_delta));
                                     ^
external/org_tensorflow/tensorflow/core/platform/macros.h:64:30: note: in definition of macro 'TF_PREDICT_FALSE'
 #define TF_PREDICT_FALSE(x) (x)
                              ^
./dragnn/core/beam.h:115:13: note: in expansion of macro 'CHECK'
             CHECK(!isnan(score_delta));
             ^
dragnn/core/beam_test.cc: In member function 'virtual void syntaxnet::dragnn::BeamTest_AdvancingCreatesNewTransitions_Test::TestBody()':
dragnn/core/beam_test.cc:177:21: warning: comparison between signed and unsigned integer expressions [-Wsign-compare]
   for (int i = 0; i < beam.beam().size(); ++i) {
                     ^
dragnn/core/beam_test.cc: In member function 'virtual void syntaxnet::dragnn::BeamTest_MultipleElementBeamsAdvanceAllElements_Test::TestBody()':
dragnn/core/beam_test.cc:245:21: warning: comparison between signed and unsigned integer expressions [-Wsign-compare]
   for (int i = 0; i < beam.beam().size(); ++i) {
                     ^
dragnn/core/beam_test.cc: In member function 'virtual void syntaxnet::dragnn::BeamTest_AdvancingDropsLowValuePredictions_Test::TestBody()':
dragnn/core/beam_test.cc:328:21: warning: comparison between signed and unsigned integer expressions [-Wsign-compare]
   for (int i = 0; i < beam.beam().size(); ++i) {
                     ^
dragnn/core/beam_test.cc: In member function 'virtual void syntaxnet::dragnn::BeamTest_AdvancesFromOracleWithMultipleStates_Test::TestBody()':
dragnn/core/beam_test.cc:406:21: warning: comparison between signed and unsigned integer expressions [-Wsign-compare]
   for (int i = 0; i < beam.beam().size(); ++i) {
                     ^
dragnn/core/beam_test.cc:413:21: warning: comparison between signed and unsigned integer expressions [-Wsign-compare]
   for (int i = 0; i < beam.beam().size(); ++i) {
                     ^
dragnn/core/beam_test.cc: In member function 'virtual void syntaxnet::dragnn::BeamTest_IgnoresForbiddenTransitionActions_Test::TestBody()':
dragnn/core/beam_test.cc:522:21: warning: comparison between signed and unsigned integer expressions [-Wsign-compare]
   for (int i = 0; i < beam.beam().size(); ++i) {
                     ^
dragnn/core/beam_test.cc: In member function 'virtual void syntaxnet::dragnn::BeamTest_FindPreviousIndexTracesHistory_Test::TestBody()':
dragnn/core/beam_test.cc:648:21: warning: comparison between signed and unsigned integer expressions [-Wsign-compare]
   for (int i = 0; i < beam.beam().size(); ++i) {
                     ^
In file included from dragnn/core/beam_test.cc:1:0:
./dragnn/core/beam.h: In instantiation of 'void syntaxnet::dragnn::Beam<T>::Init(std::vector<std::unique_ptr<T> >) [with T = syntaxnet::dragnn::{anonymous}::TestTransitionState]':
dragnn/core/beam_test.cc:117:30:   required from here
./dragnn/core/beam.h:58:23: warning: comparison between signed and unsigned integer expressions [-Wsign-compare]
     for (int i = 0; i < beam_.size(); ++i) {
                       ^
./dragnn/core/beam.h: In instantiation of 'void syntaxnet::dragnn::Beam<T>::AdvanceFromPrediction(const float*, int, int) [with T = syntaxnet::dragnn::{anonymous}::TestTransitionState]':
dragnn/core/beam_test.cc:118:66:   required from here
./dragnn/core/beam.h:100:39: warning: comparison between signed and unsigned integer expressions [-Wsign-compare]
       for (int beam_idx = 0; beam_idx < beam_.size(); ++beam_idx) {
                                       ^
In file included from external/org_tensorflow/tensorflow/core/platform/default/logging.h:24:0,
                 from external/org_tensorflow/tensorflow/core/platform/logging.h:25,
                 from external/org_tensorflow/tensorflow/core/lib/core/status.h:24,
                 from ./syntaxnet/base.h:24,
                 from ./dragnn/core/interfaces/transition_state.h:7,
                 from ./dragnn/core/interfaces/cloneable_transition_state.h:7,
                 from ./dragnn/core/beam.h:8,
                 from dragnn/core/beam_test.cc:1:
./dragnn/core/beam.h:115:25: error: 'isnan' was not declared in this scope
             CHECK(!isnan(score_delta));
                         ^
external/org_tensorflow/tensorflow/core/platform/macros.h:64:30: note: in definition of macro 'TF_PREDICT_FALSE'
 #define TF_PREDICT_FALSE(x) (x)
                              ^
./dragnn/core/beam.h:115:13: note: in expansion of macro 'CHECK'
             CHECK(!isnan(score_delta));
             ^
./dragnn/core/beam.h:115:25: note: suggested alternatives:
             CHECK(!isnan(score_delta));
                         ^
external/org_tensorflow/tensorflow/core/platform/macros.h:64:30: note: in definition of macro 'TF_PREDICT_FALSE'
 #define TF_PREDICT_FALSE(x) (x)
                              ^
./dragnn/core/beam.h:115:13: note: in expansion of macro 'CHECK'
             CHECK(!isnan(score_delta));
             ^
In file included from /usr/include/c++/5/random:38:0,
                 from /usr/include/c++/5/bits/stl_algo.h:66,
                 from /usr/include/c++/5/algorithm:62,
                 from ./dragnn/core/beam.h:4,
                 from dragnn/core/beam_test.cc:1:
/usr/include/c++/5/cmath:641:5: note:   'std::isnan'
     isnan(_Tp __x)
     ^
/usr/include/c++/5/cmath:641:5: note:   'std::isnan'
In file included from dragnn/core/beam_test.cc:1:0:
./dragnn/core/beam.h: In instantiation of 'void syntaxnet::dragnn::Beam<T>::AdvanceFromOracle() [with T = syntaxnet::dragnn::{anonymous}::TestTransitionState]':
dragnn/core/beam_test.cc:356:26:   required from here
./dragnn/core/beam.h:166:23: warning: comparison between signed and unsigned integer expressions [-Wsign-compare]
     for (int i = 0; i < beam_.size(); ++i) {
                       ^
In file included from external/org_tensorflow/tensorflow/core/platform/default/logging.h:24:0,
                 from external/org_tensorflow/tensorflow/core/platform/logging.h:25,
                 from external/org_tensorflow/tensorflow/core/lib/core/status.h:24,
                 from ./syntaxnet/base.h:24,
                 from ./dragnn/core/interfaces/transition_state.h:7,
                 from ./dragnn/core/interfaces/cloneable_transition_state.h:7,
                 from ./dragnn/core/beam.h:8,
                 from dragnn/core/beam_test.cc:1:
./dragnn/core/beam.h: In instantiation of 'int syntaxnet::dragnn::Beam<T>::FindPreviousIndex(int, int) const [with T = syntaxnet::dragnn::{anonymous}::TestTransitionState]':
dragnn/core/beam_test.cc:676:3:   required from here
./dragnn/core/beam.h:225:24: warning: comparison between signed and unsigned integer expressions [-Wsign-compare]
       CHECK(num_steps_ == beam_index_history_.size() - 1);
                        ^
external/org_tensorflow/tensorflow/core/platform/macros.h:64:30: note: in definition of macro 'TF_PREDICT_FALSE'
 #define TF_PREDICT_FALSE(x) (x)
                              ^
./dragnn/core/beam.h:225:7: note: in expansion of macro 'CHECK'
       CHECK(num_steps_ == beam_index_history_.size() - 1);
       ^
In file included from /usr/include/c++/5/algorithm:62:0,
                 from ./dragnn/core/beam.h:4,
                 from dragnn/core/beam_test.cc:1:
/usr/include/c++/5/bits/stl_algo.h: At global scope:
/usr/include/c++/5/bits/stl_algo.h:4718:5: error: 'void std::sort(_RAIter, _RAIter, _Compare) [with _RAIter = __gnu_cxx::__normal_iterator<syntaxnet::dragnn::Beam<syntaxnet::dragnn::{anonymous}::TestTransitionState>::Transition*, std::vector<syntaxnet::dragnn::Beam<syntaxnet::dragnn::{anonymous}::TestTransitionState>::Transition, std::allocator<syntaxnet::dragnn::Beam<syntaxnet::dragnn::{anonymous}::TestTransitionState>::Transition> > >; _Compare = syntaxnet::dragnn::Beam<T>::AdvanceFromPrediction(const float*, int, int) [with T = syntaxnet::dragnn::{anonymous}::TestTransitionState]::<lambda(const syntaxnet::dragnn::Beam<syntaxnet::dragnn::{anonymous}::TestTransitionState>::Transition&, const syntaxnet::dragnn::Beam<syntaxnet::dragnn::{anonymous}::TestTransitionState>::Transition&)>]', declared using local type 'syntaxnet::dragnn::Beam<T>::AdvanceFromPrediction(const float*, int, int) [with T = syntaxnet::dragnn::{anonymous}::TestTransitionState]::<lambda(const syntaxnet::dragnn::Beam<syntaxnet::dragnn::{anonymous}::TestTransitionState>::Transition&, const syntaxnet::dragnn::Beam<syntaxnet::dragnn::{anonymous}::TestTransitionState>::Transition&)>', is used but never defined [-fpermissive]
     sort(_RandomAccessIterator __first, _RandomAccessIterator __last,
     ^

```

**Comment by markomernick** [comment]
Hi @anhphamm, please try pulling down the repo again- this seems to be the same bug fixed by @danieldk in #1229.

**Comment by anhphamm** [comment]
Thank you @markomernick for pointed it out. All test passed
